libeos-payg/manager.c

```diff
@@ -857,16 +872,6 @@ epg_manager_add_code (EpgProvider   *provider,
   /* Reset the rate limiting history, since the code was successful. */
   clear_rate_limiting (self);
 
-  /* Kick off an asynchronous save.
```

Comment:
The commit message says:

> This commit also removes the
> save_state_async() call in add_code(), because I think the only time the
> state save is necessary is when the added code affected the expiry time.

The state also includes the list of used codes (fine, any new code being added will also affect the expiry time) and, I believe, the rate limit end time. Currently it looks like the latter doesn't cause the state to be flushed to disk, which is (I think) a bug, but it's an existing one since that's checked before this point.

Reply:
The rate limit end time is not part of the state that's saved to disk, as far as I can tell. I'll look at adding back the save on `add_code()`, hopefully without doing multiple identical disk writes in a row.